src/frontend/packages/lib_components/src/data/sideEffects/initiateUpload/index.ts

```diff
@@ -33,6 +36,10 @@ export const initiateUpload = async (
   );
 
   if (!response.ok) {
+    if (response.status === 400) {
+      const message = ((await response.json()) as SerializationFailure)['size'];
+      throw new Error(`${message[0]}`);
```

Comment:
you are not 100% sure that the `size` key is present in the response in case of `400` error. 
IMO you should throw something like this:
```js
const error = ((await response.json()) as SerializationFailure)['size'];
throw new Error({type: "InitiateUploadError", error});
```

Then, where you catch the error, you can check the `type` key and assert the `size` key is present in the `error` dict.

WDYT ?

Reply:
Completely agree. I was looking for a better way to catch the error in the specific case of a size error and yours seems nice.

Reply:
Only problem is I'm not sure we can pass an object to an Error constructor, but only strings.